### DISCUSSION TITLE
is there a better solution to get workday canvas kit working locally?

### DISCUSSION THREAD

**Discussion by shellwe** [Workday/canvas-kit#2353] [Q&A]
So I ran yarn and that worked, then I ran `yarn start` and it produced the error below. I was able to find a workaround by using `export NODE_OPTIONS=--openssl-legacy-provider` but I didn't know if there was a way to fix the issue when running a newer version of NPM.

yarn start
yarn run v1.22.19
$ node utils/download-fonts.js && node --max-old-space-size=2048 node_modules/.bin/start-storybook -p 9001 -c .storybook -s ./public
Downloading https://design.workdaycdn.com/beta/assets/fonts@1.0.0/roboto/ttf/Roboto-Light.ttf to /Users/myUser/Sites/personal/canvas-kit/public/Roboto-Light.ttf
Downloading https://design.workdaycdn.com/beta/assets/fonts@1.0.0/roboto/ttf/Roboto-Regular.ttf to /Users/myUser/Sites/personal/canvas-kit/public/Roboto-Regular.ttf
Downloading https://design.workdaycdn.com/beta/assets/fonts@1.0.0/roboto/ttf/Roboto-Medium.ttf to /Users/myUser/Sites/personal/canvas-kit/public/Roboto-Medium.ttf
Downloading https://design.workdaycdn.com/beta/assets/fonts@1.0.0/roboto/ttf/Roboto-Bold.ttf to /Users/myUser/Sites/personal/canvas-kit/public/Roboto-Bold.ttf
Downloading https://design.workdaycdn.com/beta/assets/fonts@1.0.0/roboto/ttf/RobotoMono-Regular.ttf to /Users/myUser/Sites/personal/canvas-kit/public/RobotoMono-Regular.ttf
info @storybook/react v6.5.9
info 
(node:99286) DeprecationWarning: --static-dir CLI flag is deprecated, see:

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated---static-dir-cli-flag
(Use `node --trace-deprecation ...` to show where the warning was created)
info => Loading presets
info => Serving static files from ././public at /
info => Loading custom manager config
info Addon-docs: using MDX1
info => Using PostCSS preset with postcss@7.0.39
info => Loading custom manager config
ℹ ｢wdm｣: wait until bundle finished: 
node:internal/crypto/hash:69
  this[kHandle] = new _Hash(algorithm, xofLen);
                  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:69:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (/Users/myUser/Sites/personal/canvas-kit/node_modules/webpack/lib/util/createHash.js:135:53)
    at NormalModule._initBuildHash (/Users/myUser/Sites/personal/canvas-kit/node_modules/webpack/lib/NormalModule.js:417:16)
    at handleParseError (/Users/myUser/Sites/personal/canvas-kit/node_modules/webpack/lib/NormalModule.js:471:10)
    at /Users/myUser/Sites/personal/canvas-kit/node_modules/webpack/lib/NormalModule.js:503:5
    at /Users/myUser/Sites/personal/canvas-kit/node_modules/webpack/lib/NormalModule.js:358:12
    at /Users/myUser/Sites/personal/canvas-kit/node_modules/loader-runner/lib/LoaderRunner.js:373:3
    at iterateNormalLoaders (/Users/myUser/Sites/personal/canvas-kit/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
    at Array.<anonymous> (/Users/myUser/Sites/personal/canvas-kit/node_modules/loader-runner/lib/LoaderRunner.js:205:4)
    at Storage.finished (/Users/myUser/Sites/personal/canvas-kit/node_modules/webpack/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55:16)
    at /Users/myUser/Sites/personal/canvas-kit/node_modules/webpack/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
    at /Users/myUser/Sites/personal/canvas-kit/node_modules/graceful-fs/graceful-fs.js:123:16
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v18.17.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by NicholasBoll** [comment]
> when running a newer version of NPM

Do you mean node? Yarn and npm are mutually exclusive. node 17 upgraded to OpenSSL 3 and this broke a lot of existing packages. Your listed solution looks to be what is required without upgrading a lot of packages that will require major releases and breaking changes.

We have an `.nvmrc` file that references node 16 which just went out of maintenance mode and will require upgrading. Once that happens, we'll have to deal with any issue node 18+ has with our current depencencies.
